packages/graph/hash_graph/lib/graph/src/types/schema/mod.rs

```diff
@@ -86,6 +86,20 @@ mod tests {
 
     use serde::{Deserialize, Serialize};
 
+    /// Will serialize as a constant value `"string"`
+    #[derive(Default, Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    pub(super) enum StringTypeTag {
+        #[default]
+        String,
+    }
+
+    #[derive(Default, Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+    #[serde(rename_all = "camelCase", deny_unknown_fields)]
+    pub struct StringTypeStruct {
+        r#type: StringTypeTag,
+    }
+
```

Comment:
But what was the issue there? It was only test data. `T` could be anything. If you don't want to use `String` for `T`, better use a type we are actually using (like `EntityTypeReference`), but I don't see a reason to add a new struct, which is only used for tests.

Reply:
Because `"items": "string"` is not a meaningful constraint in JSON schema. It's a misuse of the `items` keyword

Reply:
Using something like `*Reference` would be more distracting from the tests in my eyes. We'd then have to put in valid references and could mess up validation.

This struct is used in places like `types::schema::links::tests::maybe_ordered_array::unordered`, there is no need to check things like references because we're just checking that we're correctly de/serializing `MaybeUnorderedArray`, and an array with `"items": { "type": "string" }` is valid. `"items": "string"` is meaningless

Reply:
> it should be in the test module, but not in the regular code.

It is in the test module.

> We don't care about the content in that test.

I disagree, the entire point of the combinator structs (and any of these structs in general) are for them to **be helpful when making JSON schemas**.

Checking that 
```rust
MaybeOrderedArray {
    array: Array<"string">,
    ordered: false,
}
```

serializes to
```rust
{
    "type": "array",
    "items": "string",
    "ordered": false
}
```
Is weird, because it is _not a helpful JSON schema._ 

> If you want to give it meaning, we should enforce this

And actually yes later on I think it'd be good to stop people putting in semantically incorrect schema constraints inside the `items` field. Setting `"items": "string"` was a mistake we put in the Property Type meta-schema on the BP spec and it took over 5 hours to find.

In the mean time at least I think we should not put broken schemas inside tests that pass. This was the minimum example I could think of.

Reply:
As discussed off-GH, we will implement a proper validation for those things (later). To have the tests correct in advance, we will keep it.